website/content/labs/lab6.md

```diff
@@ -2,192 +2,220 @@
-        script:
-          - make -C labs/lab6/hello/ clean all
-        only: 
-          - master
-
-This file defines a single job called `hello:build` that builds the simple “Hello, world!” program in `lab6/hello/`. 
-
-Be careful with the colon after `hello:build`! That is the colon that goes after the name of the job (and is not part of the job name itself).
-
-The options we’ve specified in the job are the following:
-
-`stage` defines the job stage. The possible stages are `build`, `test`, and `deploy`. These are the stages that are part of the software development process! We have already covered building and testing in previous labs, and we will cover deployment next week. Using `stage` lets you group jobs into different stages, and jobs that are part of the same stage are executed in parallel. If `stage` isn’t specified, it is set to `test` by default.
-
-`script` specifies the list of commands that will be run as part of this job. Think of this as the commands you would type into the shell if you had to build this program. Note that, in this case, we’re only running one command, but we could specify multiple commands to run in sequence like this:
+In most CI systems, we will want to specify a series of actions that need to happen automatically whenever any commits are pushed to the repository. For example, we may need to build the code in our repository, and run some tests. These actions are typically referred to as *jobs*. A CI system will keep a record of all the jobs it runs, and will typically notify you if a job fails.
+
+When we push new commits to the repository, this may result in some or all of these jobs being run, which Travis CI refers to as “a build”. This is potentially confusing because, strictly speaking, “building” refers just to building/making your code but, in the context of CI, “a build” usually refers to “all the jobs that are run when you push any new commits to the repository”. 
+
+On Travis, the jobs are specified in a text file named `.travis.yml` which is placed in the root of your repository. 
+
+{{% note %}}
+The `.travis.yml` is a YAML file. YAML is a text-based file format that is commonly used for configuration files. We’ll be showing the syntax you need in this lab, but if you are curious there is a ton of documentation out there on it. Fun fact, it’s short for “YAML Ain't Markup Language”)
+{{% /note %}}
+
+We'll start by looking at a simple Travis CI file:
+
+    os: linux
+    
+    dist: xenial
+    
+    language: c
+    
+    before_install:
+        - sudo add-apt-repository -y ppa:snaipewastaken/ppa
+        - sudo apt-get update
+    
+    install:
+        - sudo apt-get install -y criterion criterion-dev
+    
+    before_script:
+        - cd libgeometry
+        - make clean
+    
+    script: 
+        - make all
+        - make tests
+        - tests/test-libgeometry
+
+Save this as a `.travis.yml` file in the root of your repository; commit it, but don't push it just yet.
```

Comment:
Not sure if students will have an issue with this, but might be a good idea to remind them of `ls -la` to see files starting with "." on Linux

Reply:
Added a note on this.